Extend gravity report docs

bernardjkim opened the pull request:
## Description
<!--Required. Provide high-level overview of what the change is for.-->
This PR updates the `gravity report` docs. Adds additional information on what information is collected and contained in the report tarball. Also adds some examples on using the lower level `gravity system report` command.

## Type of change
<!--Required. Keep only those that apply.-->

* Documentation

## Linked tickets and other PRs
<!--Required. Keep only those that apply.-->

<!--This PR addresses the following issues.-->
* Refs https://github.com/gravitational/gravity/issues/2102

## TODOs
<!--Required. Keep only those that apply and check them off as they get completed.-->

- [x] Address review feedback

bernardjkim committed:
Extend gravity report docs

bernardjkim commented:
@r0mant 
The gravity documentation already contains a section on using the `gravity report` command. I've added a bit more information on what logs and information is collected in the tarball.

Are there any specific example use cases that you think would be helpful to add to the docs?

a-palchikov commented on docs/7.x/cluster.md:
```diff
@@ -1647,4 +1647,43 @@ Example:
+This command will collect diagnostics from all Cluster nodes into the specified tarball that you can then submit for
+evaluation.
+
+If you would like to examine the debug report, here is a quick overview of what information is collected and availble
```
avail**a**ble

a-palchikov commented on docs/7.x/cluster.md:
```diff
@@ -1647,4 +1647,43 @@ Example:
+for examination. First extract the tarball into a directory of your choice. Here we'll extract the tarball into a
+`report` directory with: `mkdir report && tar -zxf report.tar.gz -C report`.
+
+The contents of the report should look something like:
```
nit: similar to this:

a-palchikov commented on docs/7.x/cluster.md:
````diff
@@ -1647,4 +1647,43 @@ Example:
+└── operation_install.a27a9fa0-c5e6-4a62-973b-2893752c25b6
+```
+
+* `cluster.json` contains the JSON-encoded cluster metadata. Pretty print the contents using the python json tool:
````
hmm, we should probably be doing this automatically then

bernardjkim committed:
Address feedback

- Typos
- Format cluster.json

r0mant commented on docs/7.x/cluster.md:
````diff
@@ -1647,4 +1647,42 @@ Example:
+└── operation_install.a27a9fa0-c5e6-4a62-973b-2893752c25b6
+```
+
+* `cluster.json` contains the JSON-encoded cluster metadata.
````
Nit: I wonder if structuring this information as a table would be a bit cleaner.

r0mant reviewed (commented):
Can we also add information about collecting report from individual nodes using lower level `gravity system report` command?

bernardjkim committed:
Address feedback

- Move tarball content/description into table format.
- Add `gravity system report` examples.

a-palchikov commented on tool/gravity/cli/register.go:
```diff
@@ -723,7 +723,7 @@ func RegisterCommands(app *kingpin.Application) *Application {
 
 	g.SystemReportCmd.CmdClause = g.SystemCmd.Command("report", "collect system diagnostics and output as gzipped tarball to terminal").Hidden()
-	g.SystemReportCmd.Filter = g.SystemReportCmd.Flag("filter", "collect only specific diagnostics ('system', 'kubernetes'). Collect everything if unspecified").Strings()
+	g.SystemReportCmd.Filter = g.SystemReportCmd.Flag("filter", "collect only specific diagnostics ('system', 'kubernetes', 'etc', 'timeline', 'resources'). Collect everything if unspecified").Strings()
```
nit: we should list these in a variable that is updated whenever a new report filter type is added.

bernardjkim committed:
List filters from variable
